core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1028,9 +1028,11 @@ private void deleteInternal(LockedInodePath inodePath, boolean recursive, boolea
               MountTable.Resolution resolution = mMountTable.resolve(alluxioUriToDel);
               String ufsUri = resolution.getUri().toString();
               UnderFileSystem ufs = resolution.getUfs();
-              if (!ufs.exists(ufsUri)) {
-                LOG.warn("Deleted file does not exist in the underfs: {}", ufsUri);
-              } else if (!ufs.delete(ufsUri, true)) {
+              if (!ufs.delete(ufsUri, true)) {
+                if (!ufs.exists(ufsUri)) {
+                  LOG.warn("The file to delete does not exist in under file system: {}", ufsUri);
+                  return;
```

Comment:
Is it safe to return here (there's more logic to be done after this block right)?